servers/zms/src/main/java/com/yahoo/athenz/zms/store/jdbc/JDBCConnection.java

```diff
@@ -5603,6 +5605,43 @@ public DomainGroupMember getPrincipalGroups(String principal, String domainName)
         return groups;
     }
 
+
+    @Override
+    public boolean updatePrincipal(String principal, int newState) {
+        final String caller = "updatePrincipal";
+
+        int principalId = getPrincipalId(principal);
+        if (principalId == 0) {
+            throw notFoundError(caller, ZMSConsts.OBJECT_PRINCIPAL, principal);
+        }
+        int affectedRows;
+        try (PreparedStatement ps = con.prepareStatement(SQL_UPDATE_PRINCIPAL)) {
+            ps.setInt(1, newState);
+            ps.setInt(2, principalId);
+            affectedRows = executeUpdate(ps, caller);
+        } catch (SQLException ex) {
+            throw sqlError(ex, caller);
+        }
+        return (affectedRows > 0);
+    }
+
+    @Override
+    public List<String> getPrincipals(int queriedState) {
+        final String caller = "getPrincipals";
+        List<String> principals = new ArrayList<>();
+        try (PreparedStatement ps = con.prepareStatement(SQL_GET_PRINCIPAL)) {
+            ps.setInt(1, queriedState);
+            try (ResultSet rs = executeQuery(ps, caller)) {
+                while (rs.next()) {
+                    principals.add(rs.getString(1));
```

Comment:
prefer using the column names since it's easier to add more fields later and not worry about the order